src/main/java/uk/gov/hmcts/ccd/domain/types/FieldValidator.java

```diff
@@ -2,12 +2,6 @@
 
 import java.util.List;
 
-import com.fasterxml.jackson.databind.JsonNode;
-import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
-
 public interface FieldValidator {
-
-    List<ValidationResult> validate(final String dataFieldId,
-                                    final JsonNode dataValue,
-                                    final CaseFieldDefinition caseFieldDefinition);
+    List<ValidationResult> validate(ValidationContext validationContext);
```

Comment:
maybe it would be less confusing to introduce a new class FieldValidationContext and not mix everything inside the ValidationContext?